src/summary_writer.jl

```diff
@@ -28,28 +28,45 @@ Arguments:
 * logdir: A string. Directory where event file will be written.
 * graph: A `Graph` object.
 """
-function FileWriter(log_dir::AbstractString; graph=get_def_graph())
+function FileWriter(log_dir::AbstractString; graph=nothing)
+    if !tf.in_eager_mode() && graph===nothing
+        graph = get_def_graph()
+    end
     mkpath(log_dir)
-    path = joinpath(log_dir, "events")
-    pyo = @py_proc pywrap_tensorflow[][:EventsWriter](py_bytes($path))
-    writer = FileWriter(pyo, String(log_dir))
+    local path
+    for i in Iterators.countfrom(1)
+        path = joinpath(log_dir, "events.out.tfevents.$i")
+        isfile(path) || break
```

Comment:
```suggestion
        ispath(path) || break
```

Reply:
As I understand it
What we are actually looking for is the first available name that we can use without overwriting anything.

The normal thing to be getting in the way is old log files.

But it is also not strictly impossible the user has put a directory, a symlink, a mount point or something else there.